DataAccess/Concrete/EntityFramework/Contexts/WebNetSampleDBContext.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using WebNetSample.Entity.Concrete;
+
+namespace DataAccess.Concrete.EntityFramework.Contexts
+{
+    public class WebNetSampleDBContext : DbContext
+    {
+        protected IConfiguration Configuration { get; set; }
+
+        public DbSet<Product> Products { get; set; }
+        public DbSet<Category> Categories { get; set; }
+        public DbSet<Supplier> Suppliers { get; set; }
+
+        public WebNetSampleDBContext()
+        {
+
+        }
+
+        public WebNetSampleDBContext(DbContextOptions dbContextOptions, IConfiguration configuration) : base(dbContextOptions)
+        {
+            Configuration = configuration;
+        }
+
+        //protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        //{
+        //    if (!optionsBuilder.IsConfigured)
+        //        base.OnConfiguring(optionsBuilder.UseSqlServer(Configuration.GetConnectionString("WebNetSampleConnectionString")));
+        //}
+
+        
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(connectionString: @"Server=DESKTOP-5JKESUF\SQLEXPRESS;Database=WebNetSampleDB;Trusted_Connection=true");
+        }
+        
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            // Generally, complex softwares include different namings in DB,
+            // so that we have to indicate these rows here explicitly
+            // otherwise, it might cause many conflicts in our DB.
+            modelBuilder.Entity<Category>(a =>
```

Comment:
Let's move each entity config to separate class, it will simplify reading and make your class less overloaded and big. Take a look at my example [here](https://github.com/dotnet-fizzyy/Diploma/blob/V2/WebAPI/WebAPI/Infrastructure/Postgres/Configuration/EpicConfiguration.cs#L7). After you add this separate class with config, you just need to call one method that collects all your configs automatically, take a look at example [here](https://github.com/dotnet-fizzyy/Diploma/blob/V2/WebAPI/WebAPI/Infrastructure/Postgres/DatabaseContext.cs#L35) as well :)

Reply:
Thank you so much. I applied this to my codebase too. Very useful!